### ISSUE TITLE
Tool Tip values showing NaN in date time x-axis when disabled legends in multi series line chart with straight line

### ISSUE THREAD

**Issue by Babyshalini1995** [apexcharts/apexcharts.js#1448]
# Bug report
Tool Tip values showing NaN in date time x-axis when disabled legends in multi series line chart with straight line. I am using React with Apex Line chart data. Given below the sandbox link.

## Codepen
https://codesandbox.io/s/dawn-dream-wllul

## Explanation
When I disable legends for series with more data points and observe the tooltip for series with less data point, tooltip value shows NaN

This happens mostly with the series data points with straight line

- What is the behavior you expect?
  ToolTip values should appear.
- What is happening instead?
  Showing NaN when legends are disabled
- What error message are you getting?
   NaN values are displayed. No error message is displayed..

Any help would be appreciated if I am missing on something or if this is an issue?

Thanks,
Shalini!

**Comment by jasmynn** [comment]
You can handle this on formatter of you x or y axis. See code below:

       tooltip: {
                shared: true,
                intersect: false,
                x: { format: 'MMMM dd yyyy HH:mm' },
                y: {
                    formatter: function (val) {
                        if (typeof val !== "undefined" && val != null && isNaN(val) === false) {
                            return val.toFixed(0);
                        }
                        return null;
                    },
                },

            },

**Comment by Babyshalini1995** [comment]
@jasmynn  - Thanks for your suggestion. It may work when we need to handle undefined values.

The issue is not about handling undefined values. Even if it has values, the value is returned as undefined when legends are disabled. 

Please look at the sandbox link and disable legends and observe the tool tip. For x-axis(Date time), it has values when all the legends are there. When the legends are disabled, the value is returned as undefined, where it should return date time.

**Comment by cola119** [comment]
@junedchhipa I have the same issue. This is a critical issue that needs to be investigated as soon as possible.

**Comment by cola119** [comment]
@junedchhipa When we investigated the cause of the problem, we found that `seriesHaveSameValues` was the problem.
https://github.com/apexcharts/apexcharts.js/blob/db64a005808180e304a751f6139aaf98fc693dfe/src/modules/tooltip/Utils.js#L180-L187

example data
```js
const series = [
  { name: 'data1', data: [0, 1, 1, 3, 4, 2] },
  { name: 'data2', data: [1, 1, 1, 1, 1, 1] } // straight line
]
```

The multi-series line chart with straight-line matches `coreUtils.seriesHaveSameValues` so this if statement condition will be`false` and will be ignored. As a result, `activeIndex` always becomes `0`(initial value). In fact, it has to be `1`.

https://github.com/apexcharts/apexcharts.js/blob/db64a005808180e304a751f6139aaf98fc693dfe/src/modules/tooltip/Utils.js#L134-L145
In this situation, `let currX = Xarrays[activeIndex][0]` becomes `undefined`, all subsequent logic breaks down...
